packages/edit-site/src/components/global-styles/custom-css.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { useContext, createPortal, useState } from '@wordpress/element';
+import { TextareaControl, Button } from '@wordpress/components';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { addFilter } from '@wordpress/hooks';
+import { BlockList } from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+
+function CustomCSSControl() {
```

Comment:
the following is still at POC stage, needs loading state adding to prevent user typing before existing custom CSS has loaded, etc.